frontend/src/components/archiveSection/archiveCard.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import CSS from "csstype";
+
+import SessionService from "../../services/session.service";
+import ManoAloeService from "../../controllers/mano-aloe.service";
+import {Archive} from "../../models/archive";
+import "./archiveSection.css";
+
+interface ArchiveCardProps {
+    who: string;
+    style: CSS.Properties;
+}
+
+interface ArchiveCardState {
+    archive: Archive | null;
+}
+
+export default class ArchiveCard extends React.Component<ArchiveCardProps, ArchiveCardState> {
+    constructor(props: ArchiveCardProps,
+                private manoAloeService: ManoAloeService) {
+        super(props);
+        this.manoAloeService = new ManoAloeService();
+    }
+
+    state: ArchiveCardState = {
+        archive: null
+    };
+
+    private loadArchiveFromLocal(): void {
+        const { who } = this.props;
+        const randomArchive = SessionService.getRandomArchive(who);
+        if (randomArchive) {
+            this.setState({ archive: randomArchive });
+        } else {
+            this.manoAloeService.getAllArchives(who)
+                .then((archives: Archive[]) => {
+                    SessionService.saveArchives(archives, who);
+                    const randomArchive = SessionService.getRandomArchive(who);
+                    this.setState({ archive: randomArchive });
+                })
+                .catch((error: Error) => {
```

Comment:
Just a thought, maybe if somehow we get to this error, it sets a specific video. Maybe like their intro vid? From for error catches.

Reply:
Got it